caravel/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -1,11 +1,56 @@
 import React from 'react';
 import { Panel } from 'react-bootstrap';
+import TimeSeriesLineChart from './charts/TimeSeriesLineChart';
 
-const ChartContainer = function () {
-  return (
-    <Panel header="Chart title">
-      chart goes here
-    </Panel>
-  );
-};
-export default ChartContainer;
+export default class ChartContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      params: this.getParamsFromUrl(),
+      data: props.viz.data,
+      height: window.innerHeight,
+      label1: 'Label 1',
+    };
```

Comment:
since `getParamsFromUrl` and `formatDates` don't need access to `this` we don't need to bind this to the methods. typically we would use `bind(this)` if we were passing a method to an event callback that needed access to the components context.